DigitalLearningSolutions.Web/Views/MyAccount/_EditRegistrationNumber.cshtml

```diff
@@ -4,23 +4,23 @@
 
 @{
   var errorHasOccurred = !ViewData.ModelState.IsValid;
-  var optionNoneSelected =  @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.None;
-  var optionYesSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
-  var optionNoSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
+  var optionNoneSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.None;
+  var optionYesSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
+  var optionNoSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
   var professionalRegConditionalId = "professional-reg-conditional-Id";
 }
 
-<div class="nhsuk-form-group @(optionNoneSelected? "nhsuk-form-group--error": "")">
+<div class="nhsuk-form-group @(errorHasOccurred && optionNoneSelected ? "nhsuk-form-group--error" : "")">
   <fieldset class="nhsuk-fieldset" aria-describedby="professional-registration-hint">
     <div class="nhsuk-hint" id="professional-registration-hint">
       Do you have a Professional Registration Number?
     </div>
 
-  @if (errorHasOccurred && optionNoneSelected) {
-    <span class="nhsuk-error-message" id="no-selection-error">
-       @(!optionYesSelected? "Select an option":"")
-    </span>
-  }
+    @if (errorHasOccurred) {
+      <span class="nhsuk-error-message" id="no-selection-error">
+        <span class="nhsuk-u-visually-hidden">Error:</span> @(optionNoneSelected ? ViewData.ModelState[nameof(Model.HasProfessionalRegistrationNumber)].Errors[0].ErrorMessage : "")

```

Comment:
This is still leaving us too open to future issues if things change down the line. We shouldn't be checking `optionNoneSelected` here, the conditions should be:

- If there is an error on the `HasProfessionalRegistrationNumber` property (i.e. if `errorHasOccurred` is true, which we are checking in the `if` above), display the error message from the ViewModel
- If there is no error on the `HasProfessionalRegistrationNumber` property, add nothing here

That way if we ever want to tweak what causes an error here, we only have to change the validation in the ViewModel and the view will just pick it up.

Reply:
Yes, I agree. I have now changed the logic how we set model state. So this validation works only based on HasProfessionalRegistrationNumber property

Reply:
We're still checking the value of `optionNoneSelected` here - since we're setting `errorHasOccurred` to only be true if there is definitely an error on `HasProfessionalRegistrationNumber` I don't think we need that, can we get rid of it?

Reply:
I have removed this completely and all look ok now.